Assets/Scripts/Game/Automap.cs

```diff
@@ -2481,10 +2485,13 @@ private void RestoreStateAutomapDungeon(bool forceNotVisitedInThisRun = false)
                     {
                         Transform currentTransformModel = currentTransformElement.GetChild(indexModel);
 
-                        MeshRenderer meshRenderer = currentTransformModel.GetComponent<MeshRenderer>();
-                        if (meshRenderer)
+                        MeshRenderer[] meshRenderers = currentTransformModel.GetComponentsInChildren<MeshRenderer>();
+                        for (int i = 0; i < meshRenderers.Length; i++)
                         {
-                            UpdateMeshRendererDungeonState(ref meshRenderer, automapDungeonState, indexBlock, indexElement, indexModel, forceNotVisitedInThisRun);
+                            if (meshRenderers[i])
+                            {
+                                UpdateMeshRendererDungeonState(meshRenderers[i], automapDungeonState, indexBlock, indexElement, indexModel, forceNotVisitedInThisRun);
```

Comment:
This is just missing a "ref" for meshRenderers[i] when passed to UpdateMeshRendererState(). Otherwise no issues found, so I'll just merge and add the missing ref on my end.

Reply:
I removed it because @rossipaolo [mentoned](https://github.com/Interkarma/daggerfall-unity/pull/2002#discussion_r563820584) that it's not needed.

Reply:
Well code doesn't compile without it. To remove it, we also need to change the corresponding ref in method signature for UpdateMeshRendererDungeonState().

But yeah @rossipaolo is correct - MeshRenderer is a class, so shouldn't be any issues with removing ref from both call and method. I'm happy to test and change on my side now merged.